src/plugins/ucx/ucx_backend.cpp

```
@@ -137,9 +137,22 @@ class nixlUcxBackendReqH : public nixlBackendReqH {
         for (nixlUcxReq req : requests_) {
             const nixl_status_t ret = nixl::ucx::ucsToNixlStatus(ucp_request_check_status(req));
             if (ret == NIXL_IN_PROG) {
-                // TODO: Need process this properly.
-                // it may not be enough to cancel UCX request
                 worker_->reqCancel(req);
+                /* ucp_request_cancel() is a no-op for RMA/zcopy operations, and
+                 * ucp_request_free() does not synchronously complete internal
+                 * requests. Drain the worker until every in-flight request has
+                 * finished, so that the zcopy completion callback (ucp_memh_put)
+                 * runs while the memory handle is still valid.
```

Cppcheck (2.21.0)

[style] The function 'getBase' is never used.

(unusedFunction)